benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java

```diff
@@ -83,240 +82,290 @@
+          @Override
+          public void applyParam(ClientConfiguration config, String value) {
+            config.rpcType = STREAMING;
+          }
+        }),
+    CONNECTION_WINDOW("BYTES", "The HTTP/2 connection flow control window.",
+        "" + DEFAULT.connectionWindow,
+        new ParamApplicator() {
+          @Override
+          public void applyParam(ClientConfiguration config, String value) {
+            config.connectionWindow = parseInt(value);
+          }
+        }),
+    STREAM_WINDOW("BYTES", "The HTTP/2 per-stream flow control window.",
+        "" + DEFAULT.streamWindow,
+        new ParamApplicator() {
+          @Override
+          public void applyParam(ClientConfiguration config, String value) {
+            config.streamWindow = parseInt(value);
+          }
+        }),
+    TARGET_QPS("INT", "Average number of QPS to shoot for.", "" + DEFAULT.targetQps, true,
+        new ParamApplicator() {
+          @Override
+          public void applyParam(ClientConfiguration config, String value) {
+            config.targetQps = parseInt(value);
+          }
+        });
 
-      private final String type;
-      private final String description;
-      private final Action action;
-      private final boolean required;
+    private final String type;
+    private final String description;
+    private final String defaultValue;
+    private final boolean required;
+    private final ParamApplicator applicator;
 
-      Option(String type, String description, Action action) {
-        this(type, description, action, false);
-      }
+    ClientParam(String type, String description, String defaultValue, ParamApplicator applicator) {
+      this(type, description, defaultValue, false, applicator);
+    }
 
-      Option(String type, String description, Action action, boolean required) {
-        this.type = type;
-        this.description = description;
-        this.action = action;
-        this.required = required;
-      }
+    ClientParam(String type, String description, String defaultValue, boolean required,
+                ParamApplicator applicator) {
+      this.type = type;
+      this.description = description;
+      this.defaultValue = defaultValue;
+      this.required = required;
+      this.applicator = applicator;
+    }
 
-      @Override
-      public String toString() {
-        return name().toLowerCase();
-      }
+    @Override
+    public String getName() {
+      return name().toLowerCase();
+    }
+
+    @Override
+    public String getType() {
+      return type;
+    }
+
+    @Override
+    public String getDescription() {
+      return description;
+    }
+
+    @Override
+    public String getDefaultValue() {
+      return defaultValue;
+    }
+
+    @Override
+    public boolean isRequired() {
+      return required;
+    }
+
+    @Override
+    public void setValue(Configuration config, String value) {
+      applicator.applyParam((ClientConfiguration) config, value);
```

Comment:
Maybe just have an abstract method setClientValue that each enum implements? Individual enum values can override individual methods, so I don't see a need need for ParamApplicator.